scripts/py_matter_idl/matter_idl/data_model_xml/handlers/handlers.py

```diff
@@ -0,0 +1,507 @@
+            # <access invokePrivilege="admin" timed="true"/>
+            if "invokePrivilege" in attrs:
+                if self._command:
+                    self._command.invokeacl = StringToAccessPrivilege(
+                        attrs["invokePrivilege"])
+                else:
+                    LOGGER.warn(
+                        f"Ignoring invoke privilege for {self._struct.name}")
+
+            return BaseHandler(self.context, handled=HandledDepth.SINGLE_TAG)
+        elif name == "field":
+            field = AttributesToField(attrs)
+            self._struct.fields.append(field)
+            return FieldHandler(self.context, field)
+        else:
+            return BaseHandler(self.context)
+
+
+class CommandsHandler(BaseHandler):
+    def __init__(self, context: Context, cluster: Cluster):
+        super().__init__(context, handled=HandledDepth.SINGLE_TAG)
+        self._cluster = cluster
+
+    def GetNextProcessor(self, name: str, attrs):
+        if name == "command":
+            return CommandHandler(self.context, self._cluster, attrs)
+        elif name in {"mandatoryConform", "optionalConform"}:
+            # Nothing to tag conformance
+            return BaseHandler(self.context, handled=HandledDepth.ENTIRE_TREE)
+        else:
+            return BaseHandler(self.context)
+
+
+class DataTypesHandler(BaseHandler):
+    def __init__(self, context: Context, cluster: Cluster):
+        super().__init__(context, handled=HandledDepth.SINGLE_TAG)
+        self._cluster = cluster
+
+    def GetNextProcessor(self, name: str, attrs):
+        if name == "section":
+            # Documentation data, skipped
+            return BaseHandler(self.context, handled=HandledDepth.ENTIRE_TREE)
+        elif name == "number":
+            # Seems like a documentation of a number format
+            #
+            # TODO: actually ensure this has no meaning
+            return BaseHandler(self.context, handled=HandledDepth.ENTIRE_TREE)
+        elif name == "enum":
+            return EnumHandler(self.context, self._cluster, attrs)
+        elif name == "bitmap":
+            return BitmapHandler(self.context, self._cluster, attrs)
+        elif name == "struct":
+            return StructHandler(self.context, self._cluster, attrs)
+        else:
+            return BaseHandler(self.context)
+
+
+class ClusterHandler(BaseHandler):
+    """ Handling /cluster elements."""
+
+    def __init__(self, context: Context, idl: Idl, attrs):
+        super().__init__(context, handled=HandledDepth.SINGLE_TAG)
+        self._idl = idl
+
+        assert ("name" in attrs)
+        assert ("id" in attrs)
+
+        self._cluster = Cluster(
+            side=ClusterSide.CLIENT,
+            name=NormalizeName(attrs["name"]),
+            code=ParseInt(attrs["id"]),
+            parse_meta=context.GetCurrentLocationMeta()
+        )
+
+    def EndProcessing(self):
+        # Global things MUST be available everywhere
+        to_add = [
```

Comment:
same here - we need some kind of generic python parsing helper.  'Cause I definitely also have a PR with this info